File: Mobile-App/app/src/main/java/com/example/happyhabits/feature_application/feauture_sleep/domain/model/Sleep.kt
package com.example.happyhabits.feature_application.feauture_sleep.domain.model

import com.example.happyhabits.core.domain.model.Habit
import java.time.LocalDate

class Sleep(
    id : String,
    userId : String,
    date : LocalDate,
    var time : String,
    var quality : String
): Habit(id = id, userId = userId, date = date)

File: API-Server/Happy Habits App/Model/Sleep.cs
using MongoDB.Bson.Serialization.Attributes;

namespace Happy_Habits_App.Model
{
    public class Sleep(DateOnly date, string userId, string time, string quality) : Habit(date, userId)
    {
        [BsonElement("Time")]
        public string Time { get; set; } = time;
        [BsonElement("Quality")]
        public string Quality { get; set; } = quality;
    }
}
